bin/templates/java/org/yarp/Nodes.java.erb

```diff
@@ -43,27 +53,67 @@ public abstract class Nodes {
         }
 
         public abstract <T> T accept(AbstractNodeVisitor<T> visitor);
+
+        public abstract Node[] childNodes();
+
+        @Override
+        public String toString() {
+            return toString("");
+        }
+
+        private String toString(String indent) {
+            StringBuilder builder = new StringBuilder();
+            builder.append(indent).append(this.getClass().getSimpleName()).append('\n');
+            for (Node child : childNodes()) {
+                if (child != null) {
+                    builder.append(child.toString(indent + "  "));
+                }
+            }
+            return builder.toString();
+        }
     }
 
     <%- nodes.each do |node| -%>
 
     <%= "#{node.comment.split("\n").map { |line| "// #{line}" }.join("\n    ")}\n" if node.comment -%>
     public static final class <%= node.name -%> extends Node {
-        <%-node.params.each do |param| -%>
-        public final <%= param.java_type %> <%= param.name %>;
+        <%- node.params.each do |param| -%>
+        public final <%= param.java_type %> <%= param.name %>;<%= ' // optional' if param.class.name.start_with?('Optional') %>
```

Comment:
Would it make sense to use `Optional<T>` here?

Reply:
It means they can be `null`. We don't use Java's `Optional` because that's a lot of extra allocations and indirections.